packages/experimental/Shadow/src/Shadow.tsx

```diff
@@ -22,12 +21,101 @@ export const Shadow = stagedComponent((props: ShadowProps) => {
       }
     }
 
-    const childWithKeyShadow = React.cloneElement(child, mergeProps(child.props, { style: shadowTokenStyleSet.key }));
+    const { style: childStyle, ...restOfChildProps } = child.props;
+
+    const shadowViewStyleProps = getStylePropsForShadowViews(childStyle, props.shadowToken);
+    const innerShadowViewProps = mergeProps(restOfChildProps, shadowViewStyleProps.inner);
+    const outerShadowViewProps = mergeProps(final, shadowViewStyleProps.outer);
+
+    const childWithInnerShadow = React.cloneElement(child, innerShadowViewProps);
 
-    return <View {...mergedProps}>{childWithKeyShadow}</View>;
+    return <View {...outerShadowViewProps}>{childWithInnerShadow}</View>;
   };
 });
 
+const getStylePropsForShadowViews = memoize(getStylePropsForShadowViewsWorker);
+function getStylePropsForShadowViewsWorker(childStyleProps: ViewStyle = {}, shadowToken: ShadowToken) {
```

Comment:
Nit again: Would move declaration above usage. I guess it works though..

Reply:
I guess with the whole worker/memoize we usually keep it at bottom...

Reply:
Yeah from what I've seen the memoize stuff goes at the bottom. Will leave it as is